addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -73,33 +78,28 @@ odoo.define('website_slides.quiz', function (require) {
                 self._renderAnswers();
                 self._renderAnswersHighlighting();
                 self._renderValidationInfo();
-                new CourseJoinWidget(self, self.channel.channelId).appendTo(self.$('.o_wslides_course_join_widget'));
+                var courseJoinWidget = new CourseJoinWidget(self, self.channel.channelId,true);
+                return courseJoinWidget;
+            }).then(function(courseJoinWidget){
+                if(self.notMember && !self.publicUser){
+                    if ($(".o_wslides_course_join_widget")[0]){
```

Comment:
We try to avoid global selectors as much as possible.
You should only alter blocks inside your widget's element.

`self.$(".o_wslides_course_join_widget")` searches inside your widget and is what you should use here.